src/devices/src/virtio/net/tap.rs

```diff
@@ -0,0 +1,456 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```

Comment:
I really prefer the `crosvm` solution of a `TapT` trait, which is implemented differently in tests. That way, you can unit test your code without `CAP_NET_ADMIN`.